File: src/main/java/boj/part3/needToReview/Lim11053.java
package boj.part3.needToReview;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Lim11053 {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int[] seq = new int[n];
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		for(int i = 0; i < n; i++) {
			seq[i] = Integer.parseInt(st.nextToken());
		}
		
		int[] maxes = new int[n];
		int answer = 0;
		
		for(int i = 0; i < n; i++) {
			maxes[i] = 1;
			
			for(int j = 0; j < i; j++) {
				if(seq[i] > seq[j]) {
					int tmp = maxes[j] + 1;
					
					if(tmp > maxes[i]) {
						maxes[i] = tmp;
					}
				}
			}
			
			if(maxes[i] > answer) {
				answer = maxes[i];
			}
		}
		
		System.out.println(answer);
	}
}

File: src/main/java/boj/part7/Lim11053.kt
package boj.part7

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val seq = br.readLine().split(" ").map { it.toInt() }
    val maxes = IntArray(n) { 1 }

    for (i in 1 until n) {
        for (j in 0 until i) {
            if (seq[j] < seq[i] && maxes[j] + 1 > maxes[i]) {
                maxes[i] = maxes[j] + 1
            }
        }
    }

    println(maxes.max())
}
